src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -825,142 +953,181 @@ def resolve_imports_and_packages(
     skip_upload_on_content_match: bool = False,
     is_permanent: bool = False,
     force_inline_code: bool = False,
-) -> Tuple[str, str, str, str, str, bool]:
-    import_only_stage = (
-        unwrap_stage_location_single_quote(stage_location)
-        if stage_location
-        else session.get_session_stage(statement_params=statement_params)
-    )
-
-    upload_and_import_stage = (
-        import_only_stage
-        if is_permanent
-        else session.get_session_stage(statement_params=statement_params)
-    )
+) -> Tuple[
+    Optional[str],
+    Optional[str],
+    Optional[str],
+    Optional[str],
+    Optional[Dict[str, Tuple[Optional[str], Optional[str]]]],
+    Optional[str],
+    bool,
+]:
 
     # resolve packages
-    resolved_packages = (
-        session._resolve_packages(
-            packages, include_pandas=is_pandas_udf, statement_params=statement_params
+    if session is None:  # In case of sandbox
+        resolved_packages = resolve_packages_in_client_side_sandbox(packages=packages)
+    else:  # In any other scenario
+        resolved_packages = (
+            session._resolve_packages(
+                packages,
+                include_pandas=is_pandas_udf,
+                statement_params=statement_params,
+            )
+            if packages is not None
+            else session._resolve_packages(
+                [],
+                session._packages,
+                validate_package=False,
+                include_pandas=is_pandas_udf,
+                statement_params=statement_params,
+            )
         )
-        if packages is not None
-        else session._resolve_packages(
-            [],
-            session._packages,
-            validate_package=False,
-            include_pandas=is_pandas_udf,
-            statement_params=statement_params,
+
+    if session is not None:
+        import_only_stage = (
+            unwrap_stage_location_single_quote(stage_location)
+            if stage_location
+            else session.get_session_stage(statement_params=statement_params)
+        )
+
+        upload_and_import_stage = (
+            import_only_stage
+            if is_permanent
+            else session.get_session_stage(statement_params=statement_params)
         )
-    )
 
     # resolve imports
+    # This udf_level_imports will be used as a return value, as it contains information even if the session is None
+    udf_level_imports = {}
     if imports:
-        udf_level_imports = {}
         for udf_import in imports:
             if isinstance(udf_import, str):
-                resolved_import_tuple = session._resolve_import_path(udf_import)
+                resolved_import_tuple = (
```

Comment:
I think we can just use `snowflake.snowpark.Session._resolve_import_path` since that's a static method, we can remove the `self` pointer from the method definition.